order-service/order-service.ts

```diff
@@ -0,0 +1,238 @@
+import { Tx, Pair, Side, Position, OrderConfig } from '../types/common.js';
+import {
+  OrderType,
+  BECH32_PREFIX,
+  LocalWallet,
+  SubaccountClient,
+  Network,
+  CompositeClient,
+  IndexerClient,
+} from '@dydxprotocol/v4-client-js';
+
+interface OrderService {
+  placeLimitOrder: (
+    pair: Pair,
+    side: Side,
+    price: number,
+    size: number,
+    config: OrderConfig,
+  ) => Promise<Tx>;
+  placeMarketOrder: (
+    pair: Pair,
+    side: Side,
+    price: number,
+    size: number,
+    config: OrderConfig,
+  ) => Promise<Tx>;
+  placeLimitTakeProfitOrder: (
+    pair: Pair,
+    side: Side,
+    price: number,
+    size: number,
+    config: OrderConfig,
+  ) => Promise<Tx>;
+  placeMarketTakeProfitOrder: (
+    pair: Pair,
+    side: Side,
+    price: number,
+    size: number,
+    config: OrderConfig,
+  ) => Promise<Tx>;
+  placeLimitStopLossOrder: (
+    pair: Pair,
+    side: Side,
+    price: number,
+    size: number,
+    config: OrderConfig,
+  ) => Promise<Tx>;
+  placeMarketStopLossOrder: (
+    pair: Pair,
+    side: Side,
+    price: number,
+    size: number,
+    config: OrderConfig,
+  ) => Promise<Tx>;
+  listPositions: () => Promise<Position[]>;
+}
+
+// order factory function
+const createOrderService = (
+  mnemonic: string,
+  network: Network,
+): OrderService => ({
+  placeLimitOrder: (pair, side, price, size, config) => {
+    return new Promise<Tx>(async (resolve) => {
+      const wallet = await LocalWallet.fromMnemonic(mnemonic, BECH32_PREFIX);
```

Comment:
You should not instantiate the wallet, it's subaccount, the client, etc. in all of the functions. You should do it once in the body of `createOrderService` through closure.